src/devices/bus/interpro/sr/edge.h

```diff
@@ -0,0 +1,294 @@
+
+private:
+	u32 m_reg0;
+
+	u32 m_control;
+	u32 m_status;
+	u32 m_fifo;
+	u32 m_kernel;
+	u32 m_attention;
+
+	u32 m_ififo_lwm;
+	u32 m_ififo_hwm;
+};
+
+class edge2_processor_device_base : public device_t, public srx_card_device_base
+{
+public:
+	const char *tag() const { return device_t::tag(); }
+
+protected:
+	edge2_processor_device_base(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock);
+
+	virtual void map(address_map &map) override;
+
+	virtual void device_start() override;
+};
+
+class edge2_framebuffer_device_base : public device_t, public srx_card_device_base
+{
+public:
+	const char *tag() const { return device_t::tag(); }
+
+protected:
+	edge2_framebuffer_device_base(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock);
+
+	virtual void map(address_map &map) override;
+
+	virtual void device_start() override;
+};
+
+class edge2plus_processor_device_base : public device_t, public srx_card_device_base
+{
+public:
+	const char *tag() const { return device_t::tag(); }
+
+	void register_screen(screen_device *screen) { m_screen = screen; }
+
+protected:
+	edge2plus_processor_device_base(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock);
+
+	virtual void map(address_map &map) override;
+
+	virtual void device_start() override;
+
+	DECLARE_WRITE_LINE_MEMBER(holda);
+	DECLARE_WRITE_LINE_MEMBER(scc_irq) { if (state) m_reg0 |= SCC_INT; else m_reg0 &= ~SCC_INT; irq0(state); }
+
+	DECLARE_READ32_MEMBER(control_r) { return m_control; };
+	DECLARE_WRITE32_MEMBER(control_w);
+	DECLARE_READ32_MEMBER(status_r) { return m_status; };
+	DECLARE_WRITE32_MEMBER(status_w) { COMBINE_DATA(&m_status); }
+	DECLARE_READ32_MEMBER(kernel_r) { return m_kernel; };
+	DECLARE_WRITE32_MEMBER(kernel_w) { COMBINE_DATA(&m_kernel); m_status |= KREG_IN_FULL; } // FIXME: what clears this?
+	DECLARE_WRITE32_MEMBER(mapping_w) { COMBINE_DATA(&m_mapping); }
+	DECLARE_READ32_MEMBER(attention_r) { return m_attention; };
+	DECLARE_WRITE32_MEMBER(attention_w) { COMBINE_DATA(&m_attention); }
+
+	DECLARE_WRITE32_MEMBER(ififo_lwm_w) { COMBINE_DATA(&m_ififo_lwm); }
+	DECLARE_WRITE32_MEMBER(ififo_hwm_w) { COMBINE_DATA(&m_ififo_hwm); }
+
+	DECLARE_READ32_MEMBER(reg0_r) { m_reg0 ^= VBLANK; return m_reg0; }
+	//DECLARE_READ32_MEMBER(reg0_r) { logerror("reg0_r vblank %d\n", m_screen->vblank()); return (m_screen == nullptr) ? (m_reg0 & ~VBLANK) : ((m_reg0 & ~VBLANK) | (m_screen->vblank() ? VBLANK : 0)); }
+	DECLARE_WRITE32_MEMBER(reg0_w) { COMBINE_DATA(&m_reg0); }
+
+	required_device<tms3203x_device> m_dsp1;
+
+private:
+	u32 m_control;
+	u32 m_status;
+	u32 m_attention;
+	u32 m_mapping;
+	u32 m_kernel;
+
+	u32 m_ififo_lwm;
+	u32 m_ififo_hwm;
+
+	u32 m_reg0;
+
+	screen_device *m_screen;
+};
+
+class edge2plus_framebuffer_device_base : public device_t, public srx_card_device_base
+{
+public:
+		const char *tag() const { return device_t::tag(); }
```

Comment:
Why is this indented twice?  Also, you can reduce it to `using device_t::tag;` if your want.